pre_commit_hooks/check_pipeline_name_only_steps.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+import argparse
+import sys
+from collections.abc import Sequence
+from typing import Any
+
+import ruamel.yaml
+
+yaml = ruamel.yaml.YAML(typ="safe")
+
+
+def check_pipeline_steps(melange_cfg: dict[str, Any]) -> tuple[bool, list[str]]:
```

Comment:
`melange` requires that the main package has a pipeline, so I have created packages with name-only steps. See `cuda-stub-meta.yaml` in extra-packages as an example. That's likely a `melange` bug of course, but it's reality today.